cfme/utils/appliance/services.py

```diff
@@ -114,3 +112,19 @@ def daemon_reload(self, log_callback=None):
             unit_name='',
             log_callback=log_callback
         )
+
+
+@attr.s
+class SystemdService(BaseService):
+    unit_name = attr.ib(type=str)
+
+
+@attr.s
+class DynaService(BaseService):
+    service_name_getting_cmd = attr.ib(type=str)
```

Comment:
I think we can find a better name for this attribute.  Good call on splitting the classes here.

The unit_name can be a cached_property, to reduce the number of SSH calls to fetch this.  We can delete the cached entry on upgrades, when it may change.

I'm also considering that while its nice to be able to use the envvar to set this value, having to define the unit in this way, and fetch it via costly ssh, is _too_ costly. We might be better off just version picking it - while less dynamic, this does not change often at all.